File: Luogu/官方題單/【算法1-6】二分查找与二分答案/P_2249_深基_13_例_1_查找.py
n, m = map(int, input().split())
A = list(map(int, input().split()))
quries = list(map(int, input().split()))

def find(x):
    left, right = 0, n - 1
    while left <= right:
        mid = (left + right) // 2
        if A[mid] < x:
            left = mid + 1
        else:
            right = mid - 1
    return left + 1 if left < n and A[left] == x else -1

for i in range(m):
    print(find(quries[i]), end = " ")

File: Luogu/官方題單/【算法1-6】二分查找与二分答案/P_2249_深基_13_例_1_查找.cpp
#include <bits/stdc++.h>
using namespace std;
#define endl '\n'

int main() {
    ios::sync_with_stdio(false); cin.tie(nullptr); cout.tie(nullptr);
    int n, m;
    cin >> n >> m;
    vector<int> A(n);
    for (int i = 0; i < n; i++) {
        cin >> A[i];
    }
    while (m--) {
        int q;
        cin >> q;
        int left = 0, right = n - 1;
        while (left <= right) {
            int mid = (left + right) / 2;
            if (A[mid] < q) {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }
        cout << (left < n && A[left] == q ? left + 1 : -1) << " ";
    }
    return 0;
}